prowler/providers/gcp/gcp_provider.py

```diff
@@ -312,25 +321,33 @@ def mutelist(self) -> GCPMutelist:
 
     @staticmethod
     def setup_session(
-        credentials_file: str, service_account: str, gcp_credentials: dict = None
+        credentials_file: str,
+        service_account: str,
+        gcp_credentials: dict = None,
+        service_account_key: dict = None,
     ) -> tuple:
         """
         Setup the GCP session with the provided credentials file or service account to impersonate
 
         Args:
             credentials_file: str
-            service_account: str
+            service_account: dict
+            gcp_credentials: dict
+            service_account_key: dict
```

Comment:
Please add more details to each argument.

Reply:
Done